Circuit/Components/BipolarJunctionTransistor.cs

```diff
@@ -68,8 +68,8 @@ public override void Analyze(Analysis Mna)
 
             Expression Vbc = sign * (Base.V - Collector.V);
             Expression Vbe = sign * (Base.V - Emitter.V);
-            Vbc = Mna.AddUnknownEqualTo(Name + "bc", Vbc);
-            Vbe = Mna.AddUnknownEqualTo(Name + "be", Vbe);
+            //Vbc = Mna.AddUnknownEqualTo(Name + "bc", Vbc);
```

Comment:
I went back and forth on these a lot, experimenting with performance and numerical stability. How did you decide these should be removed? The same question applies to the other similar cases of this (diodes, other transistor cases below).

Reply:
If I understand correctly how it works, this method adds an explicit unknown (and a row to a MNA matrix), which is useful if we want to use that value later (e.g to graph current in scope). But there's a tradeoff between performance and number of variables calculated. So my reasoning was to remove unused variables, to reduce problem size and see if simulation still works stable. I didn't found a circuit that stopped working after those changes (or i wasn't able to make it work by some component permutations) and was working fine before, but I haven't tested all example circuits, only a subset. I have a stash somewhere where I made some changes in `LiveSimulation` window to try a new permutation each time simulation diverges and if a working solution is found, then update schematic file accordingly ;)

Reply:
Your understanding is pretty much correct. However, it's not necessarily slower to add an extra unknown to the system. An unknown that has a small linear solution might have negligible extra cost, but at the same time, these variables can *help* because they are effectively a form of common subexpression elimination. 

The proper fix here is to have some kind of more explicit CSE, but in the absence of that, these extra unknowns can actually make the solution faster. The reason I left these in (vs. removed them in other cases) is because I experimentally found the solutions to be faster when I left them in there.

One minor other thing: these don't really affect what is measurable with the scope or not, the scope inserts its own variables into the system (so it doesn't depend on these).

Reply:
To summarize that comment though: I'd rather not remove these (or change them in this PR either way).

Reply:
Done.